File: algorithms/python/findtheHighestAltitude/findtheHighestAltitude.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/find-the-highest-altitude/
# Author: Miao Zhang
# Date:   2021-06-02

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        res = 0
        cur = 0
        for g in gain:
            cur += g
            res = max(res, cur)
        return res

File: algorithms/cpp/findtheHighestAltitude/findtheHighestAltitude.cpp
// Source: https://leetcode.com/problems/find-the-highest-altitude/
// Author: Miao Zhang
// Date:   2021-06-02

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int res = 0;
        int cur = 0;
        for (int& g: gain) {
            cur += g;
            res = max(res, cur);
        }
        return res;
    }
};
